storage/mdadm/trim/Makefile

```diff
@@ -0,0 +1,85 @@
+# is made available to anyone wishing to use, modify, copy, or
+# redistribute it subject to the terms and conditions of the GNU General
+# Public License v.2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# Author: guazhang <guazhang@redhat.com>
+
+# The toplevel namespace within which the test lives.
+TOPLEVEL_NAMESPACE=test-beaker
+
+# The name of the package under test:
+PACKAGE_NAME=storage
+
+# The path of the test below the package:
+RELATIVE_PATH=mdadm/trim
+
+# Version of the Test. Used with make tag.
+export TESTVERSION=0.1
+
+# The combined namespace of the test.
+export TEST=/$(TOPLEVEL_NAMESPACE)/$(PACKAGE_NAME)/$(RELATIVE_PATH)
+
+
+# A phony target is one that is not really the name of a file.
+# It is just a name for some commands to be executed when you
+# make an explicit request. There are two reasons to use a
+# phony target: to avoid a conflict with a file of the same
+# name, and to improve performance.
+.PHONY: all install download clean
+
+# executables to be built should be added here, 
+# they will be generated on the system under test.
+BUILT_FILES=
+
+# data files, .c files, scripts anything needed to either compile the test 
+# and/or run it.
+FILES=$(METADATA) runtest.sh Makefile PURPOSE main.sh
+
+run: $(FILES) build
+	./runtest.sh
+
+build: $(BUILT_FILES)
+	chmod a+x ./runtest.sh
+	chmod a+x ./main.sh
+
+clean:
+	rm -f *~ *.rpm $(BUILT_FILES)
+
+# You may need to add other targets e.g. to build executables from source code
+# Add them here:
+
+
+# Include Common Makefile
+include /usr/share/rhts/lib/rhts-make.include
+
+# Generate the testinfo.desc here:
+$(METADATA): Makefile
+	@touch $(METADATA)
+# Change to the test owner's name
+	@echo "Owner:        Changhui Zhong <czhong@redhat.com>" > $(METADATA)
+	@echo "Name:         $(TEST)" >> $(METADATA)
+	@echo "Path:         $(TEST_DIR)"       >> $(METADATA)
+	@echo "License:      GPLv3" >> $(METADATA)
+	@echo "TestVersion:  $(TESTVERSION)"    >> $(METADATA)
+	@echo "Description:  test the function trim support">> $(METADATA)
+	@echo "TestTime:     2h" >> $(METADATA)
+	@echo "RunFor:       $(PACKAGE_NAME)" >> $(METADATA)
+# add any other packages for which your test ought to run here
+	@echo "Requires:     $(PACKAGE_NAME)" >> $(METADATA)
+	@echo "Requires:	 mdadm" >> $(METADATA)
+	@echo "Requires:         iscsi-initiator-utils" >> $(METADATA)
+	@echo "Requires:         scsi-target-utils" >> $(METADATA)
+	@echo "RhtsRequires: tests-beaker-storage-mdadm-include" >> $(METADATA)
```

Comment:
I think @rasibley mentioned using `RepoRequires` for libraries. I'm not sure if this way is going to work since the include is not in the RH Beaker task library.

Reply:
Yes will need to open source this library as well so it's self contained, take a look at the networking tests and example of the metadata I mention above

Reply:
I thought it tries to point to the included library from the other files in this PR (the mdadm/include files)

Reply:
ah your right, I see it now, so just need to reference the include directory versus the beaker task, I think you can either do it in the Makefile using RepoRequires or metadata (restraint) using repoRequires, there are lots of examples in the repo.